### ISSUE TITLE
[tune] Crash with ValueError: I/O operation on closed file

### ISSUE THREAD

**Issue by hartikainen** [ray-project/ray#4744]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 18.04
- **Ray installed from (source or binary)**: binary
- **Ray version**: 0.6.5 and 0.6.6
- **Python version**: 3.6.5
- **Exact command to reproduce**:

I ran tune experiments on GCP, and on all the experiments that shared the cluster with one or more experiments crashed with the following:
```
Traceback (most recent call last):
  File "/home/ubuntu/.conda/envs/softlearning/bin/softlearning", line 11, in <module>
    load_entry_point('softlearning', 'console_scripts', 'softlearning')()
  File "/home/ubuntu/softlearning/softlearning/scripts/console_scripts.py", line 202, in main
    return cli()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/ubuntu/softlearning/softlearning/scripts/console_scripts.py", line 95, in run_example_cluster_cmd
    run_example_cluster(example_module_name, example_argv)
  File "/home/ubuntu/softlearning/examples/instrument.py", line 283, in run_example_cluster
    reuse_actors=True)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/tune/tune.py", line 235, in run
    runner.step()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/tune/trial_runner.py", line 273, in step
    self._process_events()  # blocking
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/tune/trial_runner.py", line 439, in _process_events
    self._process_trial(trial)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/tune/trial_runner.py", line 504, in _process_trial
    trial, error=True, error_msg=error_msg)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/tune/ray_trial_executor.py", line 233, in stop_trial
    trial, error=error, error_msg=error_msg, stop_logger=stop_logger)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/tune/ray_trial_executor.py", line 160, in _stop_trial
    trial.close_logger()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/tune/trial.py", line 391, in close_logger
    self.result_logger.close()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/tune/logger.py", line 233, in close
    self._log_syncer.sync_now(force=False)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/tune/log_sync.py", line 222, in sync_now
    final_cmd, shell=True, stdout=self.logfile)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/subprocess.py", line 667, in __init__
    errread, errwrite) = self._get_handles(stdin, stdout, stderr)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/subprocess.py", line 1184, in _get_handles
    c2pwrite = stdout.fileno()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/tempfile.py", line 483, in func_wrapper
    return func(*args, **kwargs)
ValueError: I/O operation on closed file
```

**Comment by richardliaw** [comment]
Hmm, are they using the same logfile?

**Comment by hartikainen** [comment]
I think that's possible. I don't set any logfiles manually anywhere.

**Comment by hartikainen** [comment]
Noting that this just happened on a couple of experiments that were running on individually on separate clusters. This happened on `ray==0.6.5`.

Here are some exceptions I see in the error logs:
```
(softlearning) ubuntu@ray-perturbations-pool-hopper-head-3eaf6c62:~/softlearning$ cat /tmp/ray/session_2019-05-11_17-19-13_12421/logs/*err | grep -B 50 -A 50 'exce'
I0511 18:31:23.176470 12480 node_manager.cc:1954] Resubmitting task 0000000014c377cc9fa5ec3d424952ac398b0bd9 on client 3daa80bacf2daf04a73e0c7ac2b3595a13386917
I0511 18:32:04.939327 12480 node_manager.cc:1954] Resubmitting task 00000000be1eefd26ddc18789edf03414c9789cd on client 3daa80bacf2daf04a73e0c7ac2b3595a13386917
I0511 18:32:15.839365 12480 node_manager.cc:1954] Resubmitting task 0000000063e82938c529b65de62c20494e3dad3c on client 3daa80bacf2daf04a73e0c7ac2b3595a13386917
I0511 18:32:15.839663 12480 node_manager.cc:1954] Resubmitting task 00000000453007492aeecb3f4d33f724f1ecd529 on client 3daa80bacf2daf04a73e0c7ac2b3595a13386917
I0511 18:34:33.357933 12480 node_manager.cc:365] [ConnectClient] Trying to connect to client cd220e9d438760b8423a46d54a15fb3a64dac06f at 10.138.0.93:35491
I0511 18:39:37.142904 12480 node_manager.cc:443] [HeartbeatAdded]: received heartbeat from unknown client id cd220e9d438760b8423a46d54a15fb3a64dac06f
I0511 18:42:12.097322 12480 node_manager.cc:443] [HeartbeatAdded]: received heartbeat from unknown client id 9f92fffaac5fc8818a8bd578e4b25f9adbe5d4f4
I0511 18:47:00.851007 12480 node_manager.cc:443] [HeartbeatAdded]: received heartbeat from unknown client id a43155f341cecadec4630e7f0e681c1e1478239a
I0511 18:47:41.709197 12480 node_manager.cc:443] [HeartbeatAdded]: received heartbeat from unknown client id c5c3c6a0527bcb25cd50336ebbbfee6c09d9b5aa
I0511 18:48:19.360077 12480 node_manager.cc:443] [HeartbeatAdded]: received heartbeat from unknown client id a2b705bdb6f373c4007bf6cc3440c6fb1b88db56
I0511 18:49:11.528044 12480 node_manager.cc:443] [HeartbeatAdded]: received heartbeat from unknown client id fc4d2f9bc8cf439ea12aa355bcddea462ff06b88
I0511 18:49:43.371343 12480 node_manager.cc:443] [HeartbeatAdded]: received heartbeat from unknown client id 17ead8403978608870c8beb6b4ffea1e2cd11507
I0511 18:50:20.621728 12480 node_manager.cc:443] [HeartbeatAdded]: received heartbeat from unknown client id 6571e9c6bbbac0e9fe77bb6c7549959933980765
WARNING: Logging before InitGoogleLogging() is written to STDERR
W0511 17:24:27.290978 12478 monitor.cc:49] Client timed out: afff869f44a20c48b34b2bfaec6b42eef8fc1481
W0511 17:25:29.446432 12478 monitor.cc:49] Client timed out: b961fb88ef30731b035af3f2c440aed2b4631c98
W0511 17:25:41.772639 12478 monitor.cc:49] Client timed out: 050ddf271dd81874b5df105b0459a11af209c5fb
W0511 17:26:36.726929 12478 monitor.cc:49] Client timed out: 4da52c8dab5334d8dffa323669b127c01e98df61
W0511 17:26:48.762661 12478 monitor.cc:49] Client timed out: 7809b37c94181472936745d2e00593b4d8f71d80
W0511 17:27:18.738600 12478 monitor.cc:49] Client timed out: f42042d6945f0487c6c8a44aa6e47bf6bb5f925d
W0511 17:27:24.954279 12478 monitor.cc:49] Client timed out: a68f22efa30851dc59cdba7546bd17b05f1fa926
W0511 17:27:37.333863 12478 monitor.cc:49] Client timed out: 9e8fbd85d8f6e0598eab20f694b333f85ad8e933
W0511 18:31:18.212957 12478 monitor.cc:49] Client timed out: 7b3bc8a5f341eb0e20e6e4aad7177e60ecb28f17
W0511 18:40:07.088948 12478 monitor.cc:49] Client timed out: cd220e9d438760b8423a46d54a15fb3a64dac06f
W0511 18:42:42.047369 12478 monitor.cc:49] Client timed out: 9f92fffaac5fc8818a8bd578e4b25f9adbe5d4f4
W0511 18:43:18.902482 12478 monitor.cc:49] Client timed out: 1156b17040f48a006a8c6dbb65619f598017eb29
W0511 18:43:55.858821 12478 monitor.cc:49] Client timed out: c0a1347014e96688e4f82489a8814f57efccc240
W0511 18:44:38.224367 12478 monitor.cc:49] Client timed out: 19d30673945f0d456c7fec8422d361ee8ad7ddd3
W0511 18:45:17.887054 12478 monitor.cc:49] Client timed out: c3a0424872a0b2718151ab71a4ceff79391a3e15
W0511 18:46:00.054332 12478 monitor.cc:49] Client timed out: 1938c6fe27c6a4940396d5a21b241e9e78437dc4
W0511 18:46:47.327090 12478 monitor.cc:49] Client timed out: ed4099a3828a335679a1d4b080dde78809dfa20a
W0511 18:47:30.791857 12478 monitor.cc:49] Client timed out: a43155f341cecadec4630e7f0e681c1e1478239a
W0511 18:48:11.649600 12478 monitor.cc:49] Client timed out: c5c3c6a0527bcb25cd50336ebbbfee6c09d9b5aa
W0511 18:48:49.297457 12478 monitor.cc:49] Client timed out: a2b705bdb6f373c4007bf6cc3440c6fb1b88db56
W0511 18:49:41.467831 12478 monitor.cc:49] Client timed out: fc4d2f9bc8cf439ea12aa355bcddea462ff06b88
W0511 18:50:13.311058 12478 monitor.cc:49] Client timed out: 17ead8403978608870c8beb6b4ffea1e2cd11507
W0511 18:50:50.567322 12478 monitor.cc:49] Client timed out: 6571e9c6bbbac0e9fe77bb6c7549959933980765
Traceback (most recent call last):
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1508, in wrapper
    return fun(self, *args, **kwargs)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1584, in name
    name = self._parse_stat_file()[0]
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_common.py", line 340, in wrapper
    return fun(self)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1547, in _parse_stat_file
    with open_binary("%s/%s/stat" % (self._procfs_path, self.pid)) as f:
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_common.py", line 582, in open_binary
    return open(fname, "rb", **kwargs)
FileNotFoundError: [Errno 2] No such file or directory: '/proc/13154/stat'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 176, in run
    self.perform_iteration()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 165, in perform_iteration
    stats = self.get_all_stats()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 156, in get_all_stats
    "workers": self.get_workers(),
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 131, in get_workers
    ]) for x in psutil.process_iter() if running_worker(x.name())
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 131, in <listcomp>
    ]) for x in psutil.process_iter() if running_worker(x.name())
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/__init__.py", line 603, in name
    name = self._proc.name()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1519, in wrapper
    raise NoSuchProcess(self.pid, self._name)
psutil._exceptions.NoSuchProcess: psutil.NoSuchProcess process no longer exists (pid=13154)
Traceback (most recent call last):
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1508, in wrapper
    return fun(self, *args, **kwargs)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1584, in name
    name = self._parse_stat_file()[0]
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_common.py", line 340, in wrapper
    return fun(self)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1548, in _parse_stat_file
    data = f.read()
ProcessLookupError: [Errno 3] No such process

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 176, in run
    self.perform_iteration()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 165, in perform_iteration
    stats = self.get_all_stats()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 156, in get_all_stats
    "workers": self.get_workers(),
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 131, in get_workers
    ]) for x in psutil.process_iter() if running_worker(x.name())
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 131, in <listcomp>
    ]) for x in psutil.process_iter() if running_worker(x.name())
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/__init__.py", line 603, in name
    name = self._proc.name()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1515, in wrapper
    raise NoSuchProcess(self.pid, self._name)
psutil._exceptions.NoSuchProcess: psutil.NoSuchProcess process no longer exists (pid=10916)
Traceback (most recent call last):
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1508, in wrapper
    return fun(self, *args, **kwargs)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1584, in name
    name = self._parse_stat_file()[0]
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_common.py", line 340, in wrapper
    return fun(self)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1547, in _parse_stat_file
    with open_binary("%s/%s/stat" % (self._procfs_path, self.pid)) as f:
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_common.py", line 582, in open_binary
    return open(fname, "rb", **kwargs)
FileNotFoundError: [Errno 2] No such file or directory: '/proc/25668/stat'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 176, in run
    self.perform_iteration()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 165, in perform_iteration
    stats = self.get_all_stats()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 156, in get_all_stats
    "workers": self.get_workers(),
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 131, in get_workers
    ]) for x in psutil.process_iter() if running_worker(x.name())
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 131, in <listcomp>
    ]) for x in psutil.process_iter() if running_worker(x.name())
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/__init__.py", line 603, in name
    name = self._proc.name()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1519, in wrapper
    raise NoSuchProcess(self.pid, self._name)
psutil._exceptions.NoSuchProcess: psutil.NoSuchProcess process no longer exists (pid=25668, name='python2')
Traceback (most recent call last):
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1508, in wrapper
    return fun(self, *args, **kwargs)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1584, in name
    name = self._parse_stat_file()[0]
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_common.py", line 340, in wrapper
    return fun(self)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1547, in _parse_stat_file
    with open_binary("%s/%s/stat" % (self._procfs_path, self.pid)) as f:
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_common.py", line 582, in open_binary
    return open(fname, "rb", **kwargs)
FileNotFoundError: [Errno 2] No such file or directory: '/proc/32480/stat'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 176, in run
    self.perform_iteration()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 165, in perform_iteration
    stats = self.get_all_stats()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 156, in get_all_stats
    "workers": self.get_workers(),
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 131, in get_workers
    ]) for x in psutil.process_iter() if running_worker(x.name())
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 131, in <listcomp>
    ]) for x in psutil.process_iter() if running_worker(x.name())
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/__init__.py", line 603, in name
    name = self._proc.name()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1519, in wrapper
    raise NoSuchProcess(self.pid, self._name)
psutil._exceptions.NoSuchProcess: psutil.NoSuchProcess process no longer exists (pid=32480, name='ssh')
Traceback (most recent call last):
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1508, in wrapper
    return fun(self, *args, **kwargs)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1584, in name
    name = self._parse_stat_file()[0]
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_common.py", line 340, in wrapper
    return fun(self)
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1547, in _parse_stat_file
    with open_binary("%s/%s/stat" % (self._procfs_path, self.pid)) as f:
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_common.py", line 582, in open_binary
    return open(fname, "rb", **kwargs)
ProcessLookupError: [Errno 3] No such process: '/proc/2657/stat'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 176, in run
    self.perform_iteration()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 165, in perform_iteration
    stats = self.get_all_stats()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 156, in get_all_stats
    "workers": self.get_workers(),
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 131, in get_workers
    ]) for x in psutil.process_iter() if running_worker(x.name())
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/ray/reporter.py", line 131, in <listcomp>
    ]) for x in psutil.process_iter() if running_worker(x.name())
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/__init__.py", line 603, in name
    name = self._proc.name()
  File "/home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/psutil/_pslinux.py", line 1515, in wrapper
    raise NoSuchProcess(self.pid, self._name)
psutil._exceptions.NoSuchProcess: psutil.NoSuchProcess process no longer exists (pid=2657)
Ray worker pid: 12489
Ray worker pid: 12487
2019-05-11 17:20:02.949856: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-05-11 17:20:03.611365: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2200000000 Hz
2019-05-11 17:20:03.651823: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x5575bce8b380 executing computations on platform Host. Devices:
2019-05-11 17:20:03.651901: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
WARNING:tensorflow:From /home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/tensorflow/python/ops/resource_variable_ops.py:435: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING: Logging before flag parsing goes to stderr.
W0511 17:20:07.059319 140135505721152 deprecation.py:323] From /home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/tensorflow/python/ops/resource_variable_ops.py:435: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/tensorflow/python/ops/math_ops.py:3066: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
W0511 17:20:08.546790 140135505721152 deprecation.py:323] From /home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/tensorflow/python/ops/math_ops.py:3066: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
WARNING:tensorflow:From /home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/tensorflow/python/ops/math_grad.py:102: div (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Deprecated in favor of operator or tf.math.divide.
W0511 17:20:08.632856 140135505721152 deprecation.py:323] From /home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/tensorflow/python/ops/math_grad.py:102: div (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Deprecated in favor of operator or tf.math.divide.
WARNING:tensorflow:From /home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/tensorflow/python/ops/losses/losses_impl.py:667: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
W0511 17:20:09.599192 140135505721152 deprecation.py:323] From /home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/tensorflow/python/ops/losses/losses_impl.py:667: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
Ray worker pid: 12488
2019-05-11 17:20:03.114598: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-05-11 17:20:03.611366: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2200000000 Hz
2019-05-11 17:20:03.651853: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x55ad3bb25380 executing computations on platform Host. Devices:
2019-05-11 17:20:03.651901: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
WARNING:tensorflow:From /home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/tensorflow/python/ops/resource_variable_ops.py:435: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING: Logging before flag parsing goes to stderr.
W0511 17:20:07.059246 139797526345536 deprecation.py:323] From /home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/tensorflow/python/ops/resource_variable_ops.py:435: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/tensorflow/python/ops/math_ops.py:3066: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
W0511 17:20:08.549938 139797526345536 deprecation.py:323] From /home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/tensorflow/python/ops/math_ops.py:3066: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
WARNING:tensorflow:From /home/ubuntu/.conda/envs/softlearning/lib/python3.6/site-packages/tensorflow/python/ops/math_grad.py:102: div (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
```

**Comment by hartikainen** [comment]
Could be something related to checkpointing, since these runs save checkpoints much more frequently that I normally do.

**Comment by richardliaw** [comment]
What type of checkpointing are you referring to?

**Comment by hartikainen** [comment]
The original `ValueError: I/O operation on closed file` might actually not be the actual cause of the failure, but just some error that happens in the restoring or something. I saw this same behavior locally after some trials failed due to a bug in the application, and only at the time of restore I saw the `ValueError: I/O operation on closed file`.

**Comment by richardliaw** [comment]
OK this happens during restoration and should be addressed in #5053
